### PR TITLE
Allow buttons to be adorned by iconography

### PR DESCRIPTION
**Pull request by kylemh**

### CODE REVIEW

File: packages/zephyr/src/Button.tsx [hunk 1]
```diff
@@ -74,9 +77,170 @@ const Loader = styled(Box)<{ shouldReduceMotion: boolean }>`
...
+const getAdornmentVariant = (hasLeft: boolean, hasRight: boolean): AdornmentVariation => {
+  if (hasLeft && hasRight) return 'both';
+  if (hasLeft && !hasRight) return 'leftOnly';
+  if (!hasLeft && hasRight) return 'rightOnly';
+  return 'none';
+};
+
+export type ButtonCompoundVariant = `${ButtonSize}_adorned-${AdornmentVariation}`;
+
+const getCompoundVariant = (
```
Comment (kylemh) [inline packages/zephyr/src/Button.tsx:91]: styled-system doesn't deal with compound variants :( (soon stitches... i am coming)

...so I did my best here.

File: packages/zephyr/src/Button.tsx [hunk 2]
```diff
@@ -74,9 +77,170 @@ const Loader = styled(Box)<{ shouldReduceMotion: boolean }>`
...
+export type AdornmentVariation = 'leftOnly' | 'rightOnly' | 'both' | 'none';
+
+const getAdornmentVariant = (hasLeft: boolean, hasRight: boolean): AdornmentVariation => {
+  if (hasLeft && hasRight) return 'both';
+  if (hasLeft && !hasRight) return 'leftOnly';
+  if (!hasLeft && hasRight) return 'rightOnly';
+  return 'none';
+};
+
+export type ButtonCompoundVariant = `${ButtonSize}_adorned-${AdornmentVariation}`;
```
Comment (kylemh) [inline packages/zephyr/src/Button.tsx:89]: 👀 template literal type in the houuuuuuseee

File: packages/zephyr/src/Button.tsx [hunk 3]
```diff
@@ -74,9 +77,170 @@ const Loader = styled(Box)<{ shouldReduceMotion: boolean }>`
...
+  'large_adorned-rightOnly': {
+    height: 48,
+    pl: 24,
+    pr: 16,
+    fontSize: 16,
+  },
+  'large_adorned-both': {
+    height: 48,
+    px: 16,
+    fontSize: 16,
```
Comment (kylemh) [inline packages/zephyr/src/Button.tsx:235]: I guess I could take fontSize and height into a size-only (non-compound) variant fn, but it just seemed like extra branching logic.

File: packages/zephyr/src/Button.tsx (hunk 2 above)
Comment (rnmp) [reply packages/zephyr/src/Button.tsx:89]: This code read like dancing somehow. Groovy.

**Review by tomaszczura — APPROVED** [review]

**Pull request merged**